internal/internal_workflow.go

```diff
@@ -1123,8 +1134,10 @@ func (d *dispatcherImpl) ExecuteUntilAllBlocked(deadlockDetectionTimeout time.Du
 			}
 		}
 		// Set allBlocked to false if new coroutines where created
-		allBlocked = allBlocked && lastSequence == d.sequence
-		if len(d.coroutines) == 0 {
+		allBlocked = allBlocked && lastSequence == d.sequence && len(d.newEagerCoroutines) == 0
+		d.coroutines = append(d.newEagerCoroutines, d.coroutines...)
+		d.newEagerCoroutines = nil
```

Comment:
This looks like it will be copying the contents of the `d.coroutines` array every time, even when `d.newEagerCoroutines` is empty. How about

```suggestion
		if len(d.newEagerCoroutines) > 0 {
			d.coroutines = append(d.newEagerCoroutines, d.coroutines...)
			d.newEagerCoroutines = nil
			allBlocked = false
		} else {
			allBlocked = allBlocked && lastSequence == d.sequence
		}
```